internal/controller/bucket_controller.go

```diff
@@ -884,6 +884,14 @@ func (r *BucketReconciler) createBucketProvider(ctx context.Context, obj *source
 		authOpts = append(authOpts, auth.WithProxyURL(*creds.proxyURL))
 	}
 
+	if obj.Spec.Region != "" {
+		authOpts = append(authOpts, auth.WithSTSRegion(obj.Spec.Region))
+	}
```

Comment:
This should be enough for setting the AWS region, it will take place for both object-level and controller-level authentication.